app/src/full/java/io/homeassistant/companion/android/sensors/LocationSensorManager.kt

```diff
@@ -765,22 +766,17 @@ class LocationSensorManager : LocationSensorManagerBase() {
         lastLocationSend = now
         lastUpdateLocation = updateLocation.gps.contentToString()
 
-        val geoSensorManager = SensorReceiver.MANAGERS.firstOrNull { it.getAvailableSensors(latestContext).any { s -> s.name == commonR.string.basic_sensor_name_geolocation } }
-        val geoSensor = AppDatabase.getInstance(latestContext).sensorDao().getFull(GeocodeSensorManager.geocodedLocation.id)
-
         ioScope.launch {
             try {
                 integrationUseCase.updateLocation(updateLocation)
                 Log.d(TAG, "Location update sent successfully")
 
                 // Update Geocoded Location Sensor
-                SensorReceiver().updateSensor(
-                    latestContext,
-                    integrationUseCase,
-                    geoSensor,
-                    geoSensorManager,
-                    GeocodeSensorManager.geocodedLocation
-                )
+                val intent = Intent(latestContext, SensorReceiverBase::class.java)
+                intent.action = SensorReceiverBase.ACTION_UPDATE_SENSOR
+                intent.putExtra("basic_sensor", GeocodeSensorManager.geocodedLocation)
```

Comment:
`"basic_sensor"` should be a constant to prevent typos, in case this function is used more in the future.

Do we really need to send the entire sensor in the intent? Wouldn't it be better to only send the ID, and handle retrieving the `BasicSensor` class in the sensor receiver?

Reply:
Ah good point, not sure why I tried to stuff the intent with everything. Will update to send only the ID to retrieve the `SensorManager` and `BasicSensor`